### PR TITLE
feat: add mailbox

### LINKED ISSUE #279: Add `Contextmenu` example
### Subject

Add `Contextmenu` example

### Description

Create a Contextmenu example with Contextmenu components.

https://yamada-ui.com/components/overlay/context-menu

### Are you willing to participate in the development of this feature and create a pull request?

None

### PR DESCRIPTION
**Pull request by illionillion**
Closes #279

### CODE REVIEW

**Review by setodeve — DISMISSED** [review]

File: contents/overlay/context-menus/mail-box/index.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+          <Text>{time}</Text>
+        </Center>
+      </ContextMenuTrigger>
+
+      <MenuList>
+        <MenuItem onClick={handleDelete}>Delete</MenuItem>
+        <MenuItem onClick={handleArchive}>
+          {isArchived ? "Remove Archive" : "Archive"}
+        </MenuItem>
+      </MenuList>
```
Comment (taroj1205) [inline contents/overlay/context-menus/mail-box/index.tsx:99]: Maybe you could add icon to menu items?
```suggestion
      <MenuList>
        <MenuItem icon={<TrashIcon color="danger" />} onClick={handleDelete}>
          Delete
        </MenuItem>
        <MenuItem
          icon={
            isArchived ? (
              <ArchiveRestoreIcon color="success" />
            ) : (
              <ArchiveIcon color="danger" />
            )
          }
          onClick={handleArchive}
        >
          {isArchived ? "Remove Archive" : "Archive"}
        </MenuItem>
      </MenuList>
```

**Review by taroj1205 — CHANGES_REQUESTED** [review]
Thanks for the PR! I have made a few suggestions, please check.

The item is active even after pressing "Archive", and I think I should be able to set it to active when clicking on the row.
It seems like the context menu does not close when I click outside, causing to open multiple context menus.

image

File: contents/overlay/context-menus/mail-box/index.tsx (hunk 1 above)
Comment (illionillion) [reply contents/overlay/context-menus/mail-box/index.tsx:99]: I accept you suggestion.

File: contents/overlay/context-menus/mail-box/index.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    </ContextMenu>
+  )
+}
```
[the remark this answers is not in the document]
Comment (illionillion) [reply contents/overlay/context-menus/mail-box/index.tsx]: Split into `mail-item.tsx`

File: contents/overlay/context-menus/mail-box/index.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    >
+      <ContextMenuTrigger
+        as={Center}
+        w="full"
+        p="md"
+        flexDir="row"
+        justifyContent="space-between"
+        borderBottomWidth="1px"
+        borderBottomStyle="solid"
+        bgColor={id === currentId ? "primary" : undefined}
```
[the remark this answers is not in the document]
Comment (illionillion) [reply contents/overlay/context-menus/mail-box/index.tsx]: How about this?

image

image

**Review by taroj1205 — APPROVED** [review]
Thanks!

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**